lib/ldp.js

```diff
@@ -44,67 +45,52 @@ const RDF_MIME_TYPES = [
  * @param [options.suppressDataBrowser]
  * @param [options.dataBrowserPath]
  * @param [options.webid]
- * @param [options.auth]
  * @param [options.multiuser]
- * @param [options.proxy]
  * @param [options.live]
- * @param [options.store]
+ * @param [options.corsProxy]
  */
 class LDP {
-  constructor (argv = {}) {
-    extend(this, argv)
+  constructor (options = {}) {
+    options = Object.assign({}, defaults, options)
+    this.auth = options.auth
+    this.serverUri = options.serverUri
+    this.dbPath = options.dbPath
+    this.configPath = options.configPath
+    this.suffixAcl = options.suffixAcl
+    this.suffixMeta = options.suffixMeta
+    this.multiuser = options.multiuser
+    this.webid = options.webid
+    this.strictOrigin = options.strictOrigin
+    this.live = options.live
+    this.errorPages = options.errorPages
+    this.noErrorPages = options.noErrorPages
+    this.errorHandler = options.errorHandler
+    this.dataBrowserPath = options.dataBrowserPath
+    this.suppressDataBrowser = options.suppressDataBrowser
 
-    // Setting root
-    if (!this.root) {
-      this.root = process.cwd()
-    }
-    if (!this.root.endsWith('/')) {
-      this.root += '/'
-    }
+    this.turtleExtensions = [ '.ttl', this.suffixAcl, this.suffixMeta ]
 
-    // Suffixes
-    if (!this.suffixAcl) {
-      this.suffixAcl = '.acl'
-    }
-    if (!this.suffixMeta) {
-      this.suffixMeta = '.meta'
+    this.root = options.root || process.cwd()
+    if (!this.root.endsWith('/')) { this.root += '/' }
+
+    if (this.corsProxy && !this.corsProxy.startsWith('/')) {
+      this.corsProxy = '/' + this.corsProxy
```

Comment:
So the above is fine, but when I code, I tend to have the rule that the length of the code should represent the complexity of the action. My suggestion would be:

```
this.root.replace(/\/*$/, '/');
this.corsProxy.replace(/$\/*/, '/');
```

Reply:
Does the regex variant take care of the `if (this.corsProxy &&` guard?

(I must admit, though I like the conciseness of regex, since i'm less familiar with it, I find the more wordy if/then more readable :) )

Reply:
Ah no, doesn't have that guard. Leave it as-is then, readability is king.